src/main/java/reposense/model/Group.java

```diff
@@ -0,0 +1,89 @@
+package reposense.model;
+
+import java.nio.file.FileSystems;
+import java.nio.file.PathMatcher;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a group in {@code RepoConfiguration}.
+ */
+public class Group {
+    public static final String DEFAULT_GROUP = "other";
+    private static final String GROUP_VALIDATION_REGEX = "[A-Za-z0-9]+";
+    private static final String MESSAGE_ILLEGAL_GROUPS = "The provided group, %s, contains illegal characters.";
+
+    private transient List<String> filePaths;
+    private transient PathMatcher groupGlobMatcher;
+    private transient String groupName;
+
+    public Group(String groupName, List<String> filePaths) {
+        validateGroup(groupName);
+        this.groupName = groupName;
+        this.filePaths = filePaths;
+        setGroupGlobMatcher();
+    }
+
+    @Override
+    public String toString() {
+        return groupName;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (this == other) {
+            return true;
+        }
+
+        // instanceof handles null
+        if (!(other instanceof Group)) {
+            return false;
+        }
+
+        Group otherGroup = (Group) other;
+        return this.groupName.equals(otherGroup.groupName);
+    }
+
+    @Override
+    public int hashCode() {
+        return groupName.hashCode();
+    }
+
+    /**
+     * Converts all the strings in {@code groups} into {@code Group} objects. Returns null if {@code groups} is null.
+     * @throws IllegalArgumentException if any of the strings are in invalid formats.
+     */
+    public static List<Group> convertStringsToGroups(List<String> groups) throws IllegalArgumentException {
```

Comment:
This seems to be used for test case and not meant to be part of main code. 
Do not mix the 2 of them, if its only used for test case, this code should be in test folder.

Reply:
I added this method to `Group`, as it is done similarly for `Format (convertStringsToFormats)`. Should I move this method to `RepoConfigurationTest` then?

Reply:
>I added this method to Group, as it is done similarly for Format (convertStringsToFormats). 

I see. Probably missed out in the reviews. All non-functional test code should be in test folder.

>Should I move this method to RepoConfigurationTest then?

Can move it to a relevant TestUtil class or create one, we might need it in other area in the future.

Reply:
Ok, i've moved this method to `TestUtil`.